plugins/modules/lvol.py

```diff
@@ -462,15 +591,20 @@ def main():
 
     if state == 'present':
         if lv_exists(module):
-            modify_lv(module, name)
+            # get initial lv properties to compare with the final
+            # state to check if something has changed with the lv
+            init_props = get_lv_props(module)
+            extend_lv(module, name, init_props)
```

Comment:
my understanding is that extendlv is to extend lv size, and chlv is to modify lv's other attributes' values. Before calling extend_lv() and modify_lv(), is it necessary to check what is going to be changed from performance perspective?  if only modifying size, then no need to call modify_lv, and varsa vise.

Reply:
i fetch the initial properties for idempotencies sake. i need to know if something changed at all so i can report properly if something or nothing changed.

since extend_lv and modify_lv are both operations to make changes on an existing lv, i put them in this one path, because one could use the module to change the size and properties at the same time.

for the cases where you ONLY JUST extend OR modify an existing LV then the corresponding run_cmd calls are skipped anyway.